src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssBuffer.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class NetSecurityNative
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct GssBuffer : IDisposable
+        {
+            internal UInt64 _length;
+            internal IntPtr _data;
+
+            internal int Copy(byte[] destination, int offset)
+            {
+                Debug.Assert(destination != null, "target destination cannot be null");
+                Debug.Assert(offset >= 0 && offset < destination.Length, "invalid offset " + offset);
```

Comment:
If `_length` is 0 (which is handled special case below) and `destination` is an empty array, `offset` could legitimately be 0 (which == `destination.Length`).  I don't know if this is really something that could happen in the program execution, but it is a case that would work fine in release which will nonetheless assert in debug.